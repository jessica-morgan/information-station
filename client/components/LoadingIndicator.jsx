import React from 'react'

const LoadingIndicator = () => {
  return (
    <div>
      <img src="/loading.gif"></img>
    </div>
  )
}

export default LoadingIndicator
