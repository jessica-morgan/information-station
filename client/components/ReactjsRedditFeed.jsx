import { useState, useEffect } from 'react'
import React, { useGlobal } from 'reactn'
import { getReactJsFeed } from '../api/redditApi'
import { selectTitle, currentTitleSelected } from '../utils'
import SingleRedditPost from './SingleRedditPost'
import LoadingIndicator from './LoadingIndicator'
import { TitlesContainer, FeedTitles } from '../styles'

const ReactjsRedditFeed = () => {
  const [posts, setPosts] = useState()

  useEffect(() => {
    getReactJsFeed()
      .then(posts => {
        let postsArray = []
        postsArray.push(posts)
        setPosts({
          allPosts: postsArray
        })
      })
  }, [])
  const [titleSelected, setTitleSelected] = useGlobal('titleSelected')
  const [categorySelected, setCategorySelected] = useGlobal('categorySelected')
  const [currentTitle, setCurrentTitle] = useGlobal('currentTitle')

  if (posts && categorySelected) {
    return <TitlesContainer>
      {posts.allPosts[0].map((singlePost, idx) => {
        return (
          <FeedTitles key={idx}>
            <h1 key={singlePost.title} onClick={() => { selectTitle(true); currentTitleSelected(singlePost.title) }}>{singlePost.title}</h1>
          </FeedTitles>
        )
      })}
    </TitlesContainer>
  } if (posts && titleSelected) {
    return <TitlesContainer>
      <SingleRedditPost posts={posts.allPosts[0]} key={currentTitle} />
    </TitlesContainer>
  } else {
    return <div>
      <LoadingIndicator />
    </div>
  }
}

export default ReactjsRedditFeed
