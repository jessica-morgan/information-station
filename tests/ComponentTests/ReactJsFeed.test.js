import React from 'reactn'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import LoadingIndicator from '../../client/components/LoadingIndicator'
import ReactjsRedditFeed from '../../client/components/ReactjsRedditFeed'

test('ReactjsRedditFeed renders correctly', () => {
  const component = renderer.create(
    <MemoryRouter>
      <ReactjsRedditFeed />
    </MemoryRouter>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('ReactjsRedditFeed should render LoadingIndicator if posts is not yet set with api data', done => {
  const wrapper = mount(<ReactjsRedditFeed />)
  setImmediate(() => {
    wrapper.update()
    expect(wrapper.containsMatchingElement(<LoadingIndicator></LoadingIndicator>)).toBeTruthy()
    done()
  })
})
