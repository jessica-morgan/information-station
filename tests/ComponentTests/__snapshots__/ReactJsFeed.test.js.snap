// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactjsRedditFeed renders correctly 1`] = `
<div>
  <div>
    <img
      className="loading-indicator"
      src="/loading.gif"
    />
  </div>
</div>
`;
